### PR TITLE
[Excel] (Range) Reorganize Range articles to improve discoverability and usability

### PR DESCRIPTION
**Pull request by alison-mk**
This PR is based on [internal work item 4453961](https://office.visualstudio.com/OC/_workitems/edit/4453961/).

**Explanation of PR** 
The PR moves all of the content from the former Ranges (fundamental) and Ranges (advanced) articles into individual articles within a "Ranges and cells" TOC sub-node. Most of the changes in the PR do not create new content. 

**New content**
- The Cells page is entirely new: docs/excel/excel-add-ins-cells.md
- All of the "new" articles according to the PR have new `title:`, `description:`, H1 titles, and intro paragraphs at the top of each article. 
- Some of the "new" articles also have new H2 elements, to help the "In this article" sidebar list populate with relevant info. 
- The "Excel cells and ranges note" include is entirely new: /includes/note-excel-cells-and-ranges.md

### CODE REVIEW

**Review by Rick-Kirkham — APPROVED** [review]
Looks good.

File: docs/excel/excel-add-ins-ranges-get.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-get.md:33]: This feels like it should link to a section on naming ranges somewhere.

File: docs/excel/excel-add-ins-ranges-get.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-get.md:50]: Suggestion for brevity.
```suggestion
The following code sample gets the used range from the worksheet named **Sample**, loads its `address` property, and writes a message to the console. The used range is the smallest range that encompasses any cells in the worksheet that have a value or formatting assigned to them. If the entire worksheet is blank, the `getUsedRange()` method returns a range that consists of only the top-left cell.
```

File: docs/excel/excel-add-ins-ranges-group.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-group.md:8]: Is it grouping ranges or the data?
```suggestion
# Group ranges for an outline using the Excel JavaScript API
```

File: docs/excel/excel-add-ins-ranges-group.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-group.md:10]: Same question as above.
```suggestion
This article provides a code sample that shows how to group ranges for an outline using the Excel JavaScript API. For the complete list of properties and methods that the `Range` object supports, see the [Excel.Range class](/javascript/api/excel/excel.range).
```

File: docs/excel/excel-add-ins-ranges-group.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-group.md:44]: Nit: extra blank line

File: docs/excel/excel-add-ins-ranges-group.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-group.md:49]: Now that this is its own article, let's add a header for ungrouping.

File: docs/excel/excel-add-ins-ranges-clear-delete.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-clear-delete.md:10]: There's a cadence to all these new article intros that's like "This article provides code samples that show..." (or something similar). Consider reframing them to give the article itself the authority, like "This article shows you how to...".

File: docs/excel/excel-add-ins-ranges-large.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-large.md:10]: Since this article isn't about a subset of Range operations, but rather a platform issue, let's drop the Range object link.
```suggestion
This article describes how to handle reading and writing to large ranges with the Excel JavaScript API.
```

File: docs/excel/excel-add-ins-ranges-large.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-large.md:14]: Let's remove the "and/or" construct.
```suggestion
If a range contains a large number of cells, values, number formats, or formulas, it may not be possible to run API operations on that range. The API will always make a best attempt to run the requested operation on a range (i.e., to retrieve or write the specified data), but attempting to perform read or write operations for a large range may result in an API error due to excessive resource utilization. To avoid such errors, we recommend that you run separate read or write operations for smaller subsets of a large range, instead of attempting to run a single read or write operation on a large range.
```

File: docs/excel/excel-add-ins-ranges-remove-duplicates.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-remove-duplicates.md:10]: Let's make sure the nuance of what's being deleted is captured here.
```suggestion
This article provides a code sample that shows how to remove duplicate entries in a range using the Excel JavaScript API. For the complete list of properties and methods that the `Range` object supports, see the [Excel.Range class](/javascript/api/excel/excel.range).
```

File: docs/excel/excel-add-ins-ranges-set-format.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-set-format.md:30]: The other articles used italics before the before-and-after images, but this one uses an H3. We should standardize this. I like an H3, but I'll defer to @lindalu-MSFT on which is preferable.

File: docs/excel/excel-add-ins-ranges-set-get-values.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-set-get-values.md:18]: Should we still include the single-cell variant in the new cell-and-range world? This could be the sample that goes in the "Cells" article.

File: docs/excel/excel-add-ins-ranges-set-get.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-set-get.md:3]:
```suggestion
description: 'Learn how to use the Excel JavaScript API to set and get ranges using the Excel JavaScript API.'
```

File: docs/excel/excel-add-ins-core-concepts.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-core-concepts.md:93]: Should the cells/ranges include file go in this article? I think it's a good intro concept that ties into the object model nicely.

File: docs/excel/excel-add-ins-ranges-string-match.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-string-match.md:2]: I'm not sure the "in a range" part is necessary. What do you think?
```suggestion
title: Find a string using the Excel JavaScript API
```

File: docs/excel/excel-add-ins-ranges-unbounded.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-unbounded.md:17]: This definition should be moved up a level, in the H1 description paragraph. It helps set context for the entire article, not just reads.

File: docs/excel/excel-add-ins-ranges-unbounded.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-unbounded.md:26]: We should add a comment to this sample that it'll fail. We don't want to lose that context.

File: docs/excel/excel-add-ins-ranges-unbounded.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-unbounded.md:23]: So what can you write to an unbounded range? This is out of scope for this PR, but we should consider fleshing this section out.

File: docs/excel/excel-add-ins-ranges-set-get.md
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-set-get.md:2]: This article is about getting and setting the selected range, so the title (and other bits) so reflect that.
```suggestion
# Set and get the selected range using the Excel JavaScript API
```

**Review by AlexJerabek — CHANGES_REQUESTED** [review]
This looks great. It dramatically increases the discoverability of this content. Thank you for doing this.

I left some comments around framing the new articles and questions about the TOC.

File: docs/excel/excel-add-ins-ranges-clear-delete.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-clear-delete.md:27]: Yes, all of the samples are existing. I didn't develop them, just copy/paste. I've tracked your suggestion in an [internal work item](https://office.visualstudio.com/OC/_workitems/edit/5008048) assigned to myself.

File: docs/toc.yml
Comment (alison-mk) [reply docs/toc.yml:683]: They follow the same order that they had within the Ranges (fundamental) and Ranges (advanced) articles, which I interpret as "easiest to hardest".

File: docs/excel/excel-add-ins-ranges-string-match.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-string-match.md:2]: I don't have a strong reason to keep it. Committing!

File: docs/excel/excel-add-ins-cells.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-cells.md:16]: I think the point of this section is to clarify that even though there's no 'Cell' class, the term cell is used in a handful of API names. 

@Rick-Kirkham Can you weigh in here?

**Comment by lindalu-MSFT** [comment]
I will review this tomorrow AM. Ran out of time today!

Thanks, Linda

**Please ignore random typos. I may have dictated this message.**

________________________________

File: docs/excel/excel-add-ins-cells.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-cells.md:42]: I didn't do this because I want to reinforce that a cell is a `Range` and all cell information should be found in `Range` articles. @Rick-Kirkham What do you think?

File: docs/excel/excel-add-ins-ranges-clear-delete.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-clear-delete.md:12]: I feel neutral about this. I think the goal is to inform anyone who might have ended up on the article without knowing that a cell is a range (perhaps via an external link). I also think readers will quickly skip over a note if they've seen it before — but that doesn't take screenreaders into account. 

@Rick-Kirkham What do you think?

File: docs/excel/excel-add-ins-ranges-get.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-get.md:33]: Do we have a section on naming ranges? I can't find one.

File: docs/excel/excel-add-ins-ranges-unbounded.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-unbounded.md:23]: I tracked your suggestion in an [internal work item](https://office.visualstudio.com/OC/_workitems/edit/5008183).

File: docs/excel/excel-add-ins-ranges-get.md
Comment (AlexJerabek) [reply docs/excel/excel-add-ins-ranges-get.md:33]: Seems like we only had the "get" part of named ranges. Let's create a low-priority work item to add the set named range conceptual section.

File: docs/excel/excel-add-ins-ranges-special-cells.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-special-cells.md:24]:
```suggestion
The following code sample uses the `getSpecialCells` method to find all the cells with formulas. About this code, note:
```

File: docs/excel/excel-add-ins-ranges-special-cells.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-special-cells.md:42]:
```suggestion
If you expect that cells with the targeted characteristic should always exist, you'll likely want your code to throw an error if those cells aren't there. If it's a valid scenario that there aren't any matching cells, your code should check for this possibility and handle it gracefully without throwing an error. You can achieve this behavior with the `getSpecialCellsOrNullObject` method and its returned `isNullObject` property. The following code sample uses this pattern. About this code, note:
```

File: docs/excel/excel-add-ins-ranges-special-cells.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-special-cells.md:44]:
```suggestion
- The `getSpecialCellsOrNullObject` method always returns a proxy object, so it's never `null` in the ordinary JavaScript sense. But if no matching cells are found, the `isNullObject` property of the object is set to `true`.
```

File: docs/excel/excel-add-ins-ranges-special-cells.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-special-cells.md:45]:
```suggestion
- It calls `context.sync` *before* it tests the `isNullObject` property. This is a requirement with all `*OrNullObject` methods and properties, because you always have to load and sync a property in order to read it. However, it's not necessary to *explicitly* load the `isNullObject` property. It's automatically loaded by the `context.sync` even if `load` is not called on the object. For more information, see [\*OrNullObject methods and properties](../develop/application-specific-api-model.md#ornullobject-methods-and-properties).
```

File: docs/excel/excel-add-ins-ranges-special-cells.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-special-cells.md:65]:
```suggestion
For simplicity, all other code samples in this article use the `getSpecialCells` method instead of  `getSpecialCellsOrNullObject`.
```

File: docs/excel/excel-add-ins-ranges-special-cells.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-special-cells.md:83]:
```suggestion
The following code sample finds special cells that are numerical constants and colors those cells pink. About this code, note:
```

File: docs/excel/excel-add-ins-ranges-special-cells.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-special-cells.md:85]:
```suggestion
- It only highlight cells that have a literal number value. It won't highlight cells that have a formula (even if the result is a number) or a boolean, text, or error state cells.
```

File: docs/excel/excel-add-ins-ranges-special-cells.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-special-cells.md:103]:
```suggestion
Sometimes you need to operate on more than one cell value type, such as all text-valued and all boolean-valued (`Excel.SpecialCellValueType.logical`) cells. The `Excel.SpecialCellValueType` enum has values with combined types. For example, `Excel.SpecialCellValueType.logicalText` targets all boolean and all text-valued cells. `Excel.SpecialCellValueType.all` is the default value, which does not limit the cell value types returned. The following code sample colors all cells with formulas that produce number or boolean value.
```

File: docs/excel/excel-add-ins-ranges-unbounded.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-ranges-unbounded.md:23]:
```suggestion
You cannot set cell-level properties such as `values`, `numberFormat`, and `formula` on an unbounded range because the input request is too large. For example, the following code example is not valid because it attempts to specify `values` for an unbounded range. The API returns an error if you attempt to set cell-level properties for an unbounded range.
```

File: docs/excel/excel-add-ins-workbooks.md
Comment (lindalu-MSFT) [inline docs/excel/excel-add-ins-workbooks.md:3]:
```suggestion
description: 'Learn how to perform common tasks with workbooks or application-level features using the Excel JavaScript API.'
```

File: docs/reference/requirement-sets/excel-api-1-10-requirement-set.md
Comment (lindalu-MSFT) [inline docs/reference/requirement-sets/excel-api-1-10-requirement-set.md:5]: I noticed that ms.prod metadata is not in the metadata of some of these. Suggest adding **ms.prod: excel** to these as well.

**Review by lindalu-MSFT — COMMENTED** [review]
LMK if you have any questions.

File: docs/excel/excel-add-ins-ranges-get.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-get.md:33]: Suggestion tracked with [internal work item](https://office.visualstudio.com/OC/_workitems/edit/5011021).

File: docs/excel/excel-add-ins-cells.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-cells.md:40]: Thank you for reviewing and for leaving so many helpful suggestions! 
In this case, I think the options for the sentence are: 
- "These ranges may be contiguous, but they don't have to be."
- "These ranges may be contiguous but don't have to be." (your suggestion, but without a comma)
Of those two options, I will stick with the first because I think removing the pronoun makes the sentence less clear.

File: docs/excel/excel-add-ins-ranges-set-get-values.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-set-get-values.md:18]: *Waiting to address this until we get feedback from Ricky.*

File: docs/excel/excel-add-ins-ranges-set-format.md
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-set-format.md:30]: I'll switch to all H3/H4 (depending on context), no italics.

**Review by AlexJerabek — APPROVED** [review]
Approving, pending LoadOptions changes.

**Pull request merged**
